pyipmi/utils.py

```diff
@@ -57,6 +57,25 @@ def check_completion_code(cc):
         raise CompletionCodeError(cc)
 
 
+def check_rsp_completion_code(rsp):
+    """
+    Check the completion code of a specific response and raise
+    CompletionCodeError in case there's an error.
+
+    This method allows to pass more metadata than the `check_completion_code`
+    method to try to interpret command-specific completion codes description in
+    case there is an error.
+
+    `rsp` should be a subclass of `Message` here.
+    """
+    if rsp.completion_code != constants.CC_OK:
+        raise CompletionCodeError(
+            rsp.completion_code,
+            cmdid=rsp.cmdid,
+            netfn=rsp.netfn,
```

Comment:
I think doing 'cmdid=rsp.cmdid & 0xfe' would be a better solution than adding the 1 in the dict. What do you think?

Reply:
Yes seems good to me